internal/controllers/extension_controller.go

```diff
@@ -116,29 +137,101 @@ func (r *ExtensionReconciler) reconcile(ctx context.Context, ext *ocv1alpha1.Ext
 		return ctrl.Result{}, nil
 	}
 
+	if !r.HasKappApis {
+		setInstalledStatusConditionFailed(&ext.Status.Conditions, errkappAPIUnavailable.Error(), ext.GetGeneration())
+
+		ext.Status.ResolvedBundleResource = ""
+		setResolvedStatusConditionUnknown(&ext.Status.Conditions, "kapp apis are unavailable", ext.GetGeneration())
```

Comment:
Technically, we _could_ resolve a bundle, figure out the correct deprecation condition values, and populate everything "normally". And then only the `Installed` condition would be `False` with this message.

At a minimum, I would suggest moving this below the `paused` check.

Reply:
I had intentionally put the check above resolution thinking that if applier is not in place to install bundle contents, then resolution is of no use. But what you said is also right. For now, I've put it below the `paused` check. Will revisit to also set the deprecation status after resolution instead of installation.